src/reporter/pegasus_counter_reporter.cpp

```diff
@@ -167,21 +225,63 @@ void pegasus_counter_reporter::update()
 
         bool first_append = true;
         _falcon_metric.timestamp = timestamp;
+
         perf_counters::instance().iterate_snapshot(
             [&oss, &first_append, this](const dsn::perf_counters::counter_snapshot &cs) {
                 _falcon_metric.metric = cs.name;
                 _falcon_metric.value = cs.value;
                 _falcon_metric.counterType = "GAUGE";
+
                 if (!first_append)
                     oss << ",";
                 _falcon_metric.encode_json_state(oss);
                 first_append = false;
             });
         oss << "]";
-
         update_counters_to_falcon(oss.str(), timestamp);
     }
 
+    if(_enable_prometheus) {
+
+        using namespace prometheus;
+        const auto labels = Gateway::GetInstanceLabel(GetHostName());
+        Gateway gateway{_prometheus_host, std::to_string(_prometheus_port), "pegasus", labels};
+        auto registry = std::make_shared<Registry>();
+
+        perf_counters::instance().iterate_snapshot(
+            [&registry, this](const dsn::perf_counters::counter_snapshot &cs) {
+                std::string metrics_name = cs.name;
+                
+                replace(metrics_name.begin(),metrics_name.end(),'@',':');
+                replace(metrics_name.begin(),metrics_name.end(),'.','_');
+                replace(metrics_name.begin(),metrics_name.end(),'*','_');
+                replace(metrics_name.begin(),metrics_name.end(),'(','_');
+                replace(metrics_name.begin(),metrics_name.end(),')','_');
+                ddebug("%s", cs.name.c_str());
+                
+                std::string app[3] = {"", "", ""};
+                std::vector<std::string> ret = s_split(metrics_name, ":");
+                if(ret.size() > 1){
+                    std::vector<std::string> ret1 = s_split(ret[1], "_");
+                    for(int i = 0; i < ret1.size(); i++){
+                        app[i] = ret1[i];
+                    }
+                }
+                auto& gauge_family_all = BuildGauge()
+                                        .Name(metrics_name)
+                                        .Labels(
+                                            {{"service","pegasus"},{"cluster",_cluster_name},{"job",_app_name},{"port",std::to_string(_local_port)}})
+                                        .Register(*registry);
+                auto& second_gauge_all = gauge_family_all.Add({{"app_id",app[0]},{"partition_count",app[1]},{"percent",app[2]}}
+                    );
+
+                second_gauge_all.Set(cs.value);
```

Comment:
I could not figure out what code from line262 to line278 are doing, you can add comments or just explain in pr description.